### PR TITLE
chore: Refactor JS action entity button checks

### PR DESCRIPTION
**Pull request by ankitakinger**
## Description

Refactoring JS action entity button checks

#### PR fixes following issue(s)
Fixes [#29762](https://github.com/appsmithorg/appsmith/issues/29762)

#### Type of change
- Chore (housekeeping or task changes that don't impact user perception)

#### How Has This Been Tested?
- [x] Manual
- [ ] JUnit
- [ ] Jest
- [x] Cypress

#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

## Summary by CodeRabbit

- **Refactor**
  - Simplified display logic for JavaScript collections by removing unused properties.
  
- **Bug Fixes**
  - Corrected the conditional display of context menus and edit icons in the JavaScript editor.

- **Style**
  - Updated user interface elements to reflect changes in display properties.

### CODE REVIEW

File: app/client/src/pages/Editor/Explorer/JSActions/JSActionEntity.tsx [hunk 1]
```diff
@@ -83,7 +83,7 @@ export const ExplorerJSCollectionEntity = memo(
         canDelete={canDeleteJSAction}
         canManage={canManageJSAction}
         className={EntityClassNames.CONTEXT_MENU}
-        hideMenuItems={jsAction?.hideContextMenu || false}
+        hideMenuItems={jsAction?.isMainJSCollection ? true : false}
```
Comment (ashit-rath) [inline app/client/src/pages/Editor/Explorer/JSActions/JSActionEntity.tsx:86]: Can we just do 
```
hideMenuItems={jsAction?.isMainJSCollection}
```

File: app/client/src/pages/Editor/JSEditor/index.tsx [hunk 2]
```diff
@@ -67,6 +67,12 @@ function JSEditor(props: Props) {
       <JsEditorForm
         backLink={isPagesPaneEnabled ? null : backLink}
         contextMenu={contextMenu}
+        hideContextMenuOnEditor={
+          jsCollectionData?.config.isMainJSCollection ? true : false
```
Comment (ashit-rath) [inline app/client/src/pages/Editor/JSEditor/index.tsx:71]: Same here
```
hideContextMenuOnEditor={jsCollectionData?.config.isMainJSCollection}
```

Comment (ankitakinger) [reply app/client/src/pages/Editor/JSEditor/index.tsx:71]: `isMainJsCollection` is not a mandatory field so I will have to make `hideContextMenuOnEditor` optional as well.

Comment (ashit-rath) [reply app/client/src/pages/Editor/JSEditor/index.tsx:71]: Would doing it like this help?
```
hideContextMenuOnEditor={Boolean(jsCollectionData?.config.isMainJSCollection)}
```

Comment (ankitakinger) [reply app/client/src/pages/Editor/JSEditor/index.tsx]: done

**Review by ashit-rath — APPROVED** [review]

**Pull request merged**